### PR TITLE
Add new collection form & display

### PR DESCRIPTION
**Pull request by joel-oe-lacey**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [X] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description
This PR adds the ability for a user to create a new collection. A user can route from the reading list page by clicking the + button. This will send them to a collection form. Once they fill out the collection form, that collection is pushed to the Back End, the user is then re-routed back to the reading list where they can see their new collection.

## Related Tickets & Documents
#27 

## Added tests?

- [ ] yes
- [ ] no, because they aren't needed
- [ ] no, because I need help
- [X] yes, more to be added

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [ ] no documentation needed
- [X] no, needs to be added

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: app/assets/stylesheets/articles.scss [hunk 1]
```diff
@@ -63,7 +63,7 @@
   .side-bar {
     float: left;
     width: 280px;
-    height: 700px;
+    height: 1000px;
```
Comment (mintona) [inline app/assets/stylesheets/articles.scss:66]: We chatted about this as a group "in-person"

File: app/controllers/api/v0/reading_collections_controller.rb [hunk 2]
```diff
@@ -5,7 +5,6 @@ def create
         reading_collection = current_user.reading_collections.create(collection_params)
         articles = reading_collection.get_articles
         reading_collection.articles << articles
-        @reading_collection = reading_collection.to_json
```
Comment (mintona) [inline app/controllers/api/v0/reading_collections_controller.rb:8]: This was a group decision as we don't need to send anything back to FE from the endpoint.

File: app/javascript/readingList/collectionForm.jsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    const tagArr = tags.split(',');
+    if (title.length && tags.length) {
+      fetch('/api/reading_collections', {
+        method: 'POST',
+        headers: {
+          Accept: 'application/json',
+          'X-CSRF-Token': window.csrfToken,
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
```
Comment (mintona) [inline app/javascript/readingList/collectionForm.jsx:36]: Nice work on getting this form done! We talked "in-person" about cleaning up how the tags are coming in on a future PR.

File: app/views/reading_collections/new.html.erb [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<% title "New Reading Collection" %>
+
+<div class="home">
+  <div class="articleformcontainer" id="collection-form">
+  </div>
+</div>
+
+<%= javascript_pack_tag "collectionForm", defer: true %>
```
Comment (mintona) [inline app/views/reading_collections/new.html.erb:8]: This makes sense! Thanks taking over the FE.

**Review by mintona — APPROVED** [review]
Nice job team on connecting the dots to link the BE and FE together.

**Review by philjdelong — APPROVED** [review]

**Pull request merged**